test/fixtures/tape-standalone.js

```diff
@@ -9,10 +9,50 @@ if (navigator.userAgent.includes("Chrome/")) {
   browser = "Firefox";
 }
 
+function getArgs(/* desc, options, fn */) {
+  let desc = "(anonymous)";
+  // By setting `timeout` to `0` here, we are allowing V8 to consider
+  // it a to have a signature of `{timeout: number}` instead of `{}`
+  // with a `timeout` property, which is easier on the C++ portion.
+  let options = {timeout: 0};
+  /** @type {function(any):void|Promise<void>} */
+  let fn;
+
+  for (let i = 0, arg; i < arguments.length; i++) {
+    arg = arguments[i];
+    switch (typeof arg) {
+      case "string":
+        desc = arg;
+        break;
+      case "function":
+        fn = arg;
+        break;
+      case "object":
+        Object.assign(options, arg);
+        break;
+    }
+  }
+
+  if (!options.timeout || options.timeout < DEFAULT_TIMEOUT) {
+    options.timeout = DEFAULT_TIMEOUT;
+  }
+
+  return {
+    desc,
+    options,
+    fn,
+  };
+}
+
 // Export as a global a wrapped test function which enforces a timeout by default.
-window.test = (desc, fn) => {
-  tape(`${desc} (${browser})`, async (t) => {
-    t.timeoutAfter(DEFAULT_TIMEOUT);
+/**
+ * @param {string} [desc]
+ * @param {object} [options]
+ * @param {function(any):void|Promise<void>} fn
+ */
+window.test = (desc, options, fn) => {
+  ({desc, options, fn} = getArgs(desc, options, fn));
```

Comment:
The implementation here is unnecessarily complicated. All you need is to take change the method signature to this:

```javascript
window.test = (desc, fn, options = {}) => {
```

and then set the default timeout, like this:

```javascript
if (!options.timeout) {
  options.timeout = DEFAULT_TIMEOUT;
}
```

The above implementation modifies the input parameter. If you want to avoid modifying the input parameter, then you could make a fresh copy as follows (instead of the `if (!options.timeout) {...}` block):

```javascript
options = {
  timeout: DEFAULT_TIMEOUT,
  ...options,
}
```

Reply:
Yeah, but&nbsp;I&nbsp;prefer putting the&nbsp;options on&nbsp;the&nbsp;second place in&nbsp;the&nbsp;method signature and allow omitting the&nbsp;description, like what tape&nbsp;does.

Reply:
We aren't using that format, and the resulting code becomes considerably more complicated with it.

Supporting only one format is easier to maintain (YAGNI), so let's stick to that.

Reply:
I prefer putting the options before the test case, as it makes it easier to find them, rather than having to find the end of the test case function.

It also makes it obvious which test cases are using custom options and which ones are not.